File: POLYGLOT QUEST 2D - 1/classes.py
from enum import Enum

## enums
class Indicators (Enum):
    NOTHING = 0

Lines = Enum('Lines', 'NOLINE SINGLELINE DOUBLELINE')
Modes = Enum ('Modes', 'NOMODE INT_FLOAT INT_FLOAT_FLOAT INT_INT_INT_INT')
Parameters = Enum('Parameters', 'THERMAL_CONDUCTIVITY HEAT_SOURCE')
Sizes = Enum('Sizes', 'NODES ELEMENTS DIRICHLET NEUMANN')
###

class item:
    id = 0
    x = 0.0
    y = 0.0
    node1 = 0
    node2 = 0
    node3 = 0
    value = 0.0

    def setId(self, id):
        self.id = id

    def setX(self, x_coordinate):
        self.x = x_coordinate

    def setY(self, y_coordinate):
        self.y = y_coordinate
    
    def setNode1(self, node1):
        self.node1 = node1
    
    def setNode2(self, node2):
        self.node2 = node2
    
    def setNode3(self, node3):
        self.node3 = node3

    def setValue(self, value):
        self.value = value

    def getId(self):
        return self.id
    
    def getX(self):
        return self.x
    
    def getY(self):
        return self.y
    
    def getNode1(self):
        return self.node1
    
    def getNode2(self):
        return self.node2
    
    def getNode3(self):
        return self.node3
    
    def getValue(self):
        return self.value
    



class Node (item):

    def setValues(self, a, b, c):
        self.id = a
        self.x = b
        self.y = c



class Element(item):
    
   def setValues(self, a, d, e, f):
       self.id = a
       self.node1 = d
       self.node2 = e
       self.node3 = f

class Condition(item):
    
    def setValues(self, d, g):
        self.node1 = d
        self.value = g

class Mesh:
    parameters = []
    sizes = []
    node_list = []
    element_list = []
    indices_dirich = []

    dirichlet_list = [] #Array containing Dirichlet objects
    neumann_list = [] #Array containing Neumann objects

    def setParameters(self, k, Q):
        self.parameters.insert(Parameters.THERMAL_CONDUCTIVITY.value - 1, k)
        self.parameters.insert(Parameters.HEAT_SOURCE.value - 1, Q)
    
    def setSizes(self, nNodes, nElemts, nDirich, nNeumn):
      
        self.sizes.insert(0, nNodes)
        self.sizes.insert(1, nElemts)
        self.sizes.insert(2, nDirich)
        self.sizes.insert(3, nNeumn)
     

    def getSize(self, s):
        return self.sizes[s]

    def getParameter(self, p):
        return self.parameters[p]

    def createData(self):
        
        for i in range(self.sizes[0]):
            obj1 = Node()
            self.node_list.append(obj1)# * self.sizes[0]
        
        for i in range(self.sizes[1]):
            obj2 = Element()
            self.element_list.append(obj2)# * self.sizes[0]
        
        for i in range(self.sizes[2]):
            obj3 = Condition()
            self.dirichlet_list.append(obj3)# * self.sizes[0]

        for i in range(self.sizes[3]):
            obj4 = Condition()
            self.neumann_list.append(obj4)# * self.sizes[0]

        self.indices_dirich = [0] * self.sizes[2] #DIRICH SIZE INDEX = 2 IN sizes Array

        

    def getNodes(self):
        return self.node_list
    
    def getElements(self):
        return self.element_list

    def getDirichletIndices(self):
        return self.indices_dirich

    def getDirichlet(self):
        return self.dirichlet_list

    def getNeumann(self):
        return self.neumann_list

    def getNode(self, i):
        return self.node_list[i]
    
    def getElement(self, i):
        return self.element_list[i]
    
    def getCondition(self, i, type):
        
        if(type == Sizes.DIRICHLET):
            return self.dirichlet_list[i]
        else:
            return self.neumann_list[i]

File: POLYGLOT QUEST - THE FINAL/PolyglotQuest3D/MEF3D/Classes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MEF3D
{
    

    public class Classes
    {
        public enum indicator { NOTHING };
        public enum line { NOLINE, SINGLELINE, DOUBLELINE };

        //TODO: MODIFIED MODE ENUM TO SAVE MORE MODES
        public enum mode { NOMODE, INT_FLOAT, INT_FLOAT_FLOAT_FLOAT, INT_INT_INT_INT_INT };
        public enum parameter { THERMAL_CONDUCTIVITY, HEAT_SOURCE };
        public enum size { NODES, ELEMENTS, DIRICHLET, NEUMANN };

        /*
        public class item
        {
            public int id { get; set; }
            public float x { get; set; }
            public float y { get; set; }
            public float z { get; set; }
            public int node1 { get; set; }
            public int node2 { get; set; }
            public int node3 { get; set; }
            public int node4 { get; set; }
            public float value { get; set; }

            public virtual void setValues(int a, float b, float c, float d, int e, int f, int g, int h, float i) { }
        }

        public class node : item
        {
            public override void setValues(int a, float b, float c, float d, int e, int f, int g, int h, float i)
            {
                id = a;
                x = b;
                y = c;
                z = d;
            }
        }

        public class element : item
        {
            public override void setValues(int a, float b, float c, float d, int e, int f, int g, int h, float i)
            {
                id = a;
                node1 = e;
                node2 = f;
                node3 = g;
                node4 = h;

            }
        }

        public class condition : item{

            public override void setValues(int a, float b, float c, float d, int e, int f, int g, int h, float i)
            {
                node1 = e;
                value = i;
            }

        }

        public class mesh
        {
            public float[]  parameters = new float [2];
            public int[] sizes = new int[4];
            public int[] indices_dirich;

            public node[] node_list;
            public element[] element_list;
            public condition[] dirichlet_list;
            public condition[] neumann_list;

            public void setParameters(float k, float Q)
            {
                parameters[(int)parameter.THERMAL_CONDUCTIVITY] = k;
                parameters[(int)parameter.HEAT_SOURCE] = Q;
            }
            public void setSizes(int nnodes, int neltos, int ndirich, int nneu)
            {
                sizes[(int)size.NODES] = nnodes;
                sizes[(int)size.ELEMENTS] = neltos;
                sizes[(int)size.DIRICHLET] = ndirich;
                sizes[(int)size.NEUMANN] = nneu;
            }
            public int getSize(int s)
            {
                return sizes[s];
            }
            public float getParameter(int p)
            {
                return parameters[p];
            }
            public void createData()
            {
                node_list = new node[(int)size.NODES];
                element_list = new element[(int)size.ELEMENTS];
                indices_dirich = new int[(int)size.DIRICHLET];
                dirichlet_list = new condition[(int)size.DIRICHLET];
                neumann_list = new condition[(int)size.NEUMANN];
            }
            public node[] getNodes()
            {
                return node_list;
            }
            public element[] getElements()
            {
                return element_list;
            }
            public int[] getDirichletIndices()
            {
                return indices_dirich;
            }
            public condition[] getDirichlet()
            {
                return dirichlet_list;
            }
            public condition[] getNeumann()
            {
                return neumann_list;
            }
            public node getNode(int i)
            {
                return node_list[i];
            }
            public element getElement(int i)
            {
                return element_list[i];
            }
            public condition getCondition(int i, int type)
            {
                if (type == (int)size.DIRICHLET) return dirichlet_list[i];
                else return neumann_list[i];
            }


        }

        */
    }
}
